games/ShootEmUp/scripts/Player.gd

```diff
@@ -43,28 +44,43 @@ func _set_invulnerable() -> void:
 	
 func take_damage(damage_amount: float = 1) -> void:
 	if not is_invulnerable:
+		impact_audio.play()
 		health -= damage_amount
-		player_take_damage.emit(health)
+		
+		# is the player dead?
 		if health <= 0:
+			await hit_stop(0.01, 1.8)
 			die()
+			
+		# player is still alive
 		else:
 			_set_invulnerable()
-			_hurt_animation()
-			hurt_audio.play()
+			await _hurt_animation()
+		player_take_damage.emit(health)
 
 func _hurt_animation() -> void:
+	await hit_stop(0.01, 1.5)
 	animation_player.set_assigned_animation("damage")
 	animation_player.play()
+	
+func hit_stop(time_scale: float, duration: float) -> void:
+	Engine.time_scale = time_scale
+	await get_tree().create_timer(time_scale * duration).timeout
+	Engine.time_scale = 1
 
 func die() -> void:
 	if not is_dead:
 		is_dead = true
-		hide()
-		death_audio.play()
+		# set the explosion pitch scale
+		var explosion_pitch_scale = randf_range(0.7, 1.5)
+		explosion_death_audio.pitch_scale = explosion_pitch_scale
```

Comment:
Since the assignment line would not be any longer, or more difficult to understand without the intermediate variable here, I think we should just remove the intermediate variable:

```suggestion
		explosion_death_audio.pitch_scale = randf_range(0.7, 1.5)
```